### PR TITLE
pgrok: init at 1.3.3

### PR DESCRIPTION
**Pull request by NyCodeGHG**
###### Description of changes

pgrok is a self hosted ngrok alternative written in go.
https://github.com/pgrok/pgrok

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/networking/pgrok/default.nix [hunk 1]
```diff
@@ -0,0 +1,37 @@
+{ buildGoModule, fetchFromGitHub, lib, nix-update-script }:
```
Comment (AndersonTorres) [inline pkgs/tools/networking/pgrok/default.nix:1]:
```suggestion
{ lib
, buildGoModule
, fetchFromGitHub
, nix-update-script 
}:

```

File: pkgs/tools/networking/pgrok/default.nix [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    "-X main.version=${version}"
+    "-X main.commit=unknown"
+    "-X main.date=unknown"
+  ];
+
+  postInstall = ''
+    moveToOutput bin/pgrokd $server
+  '';
+
+  vendorHash = "sha256-laSfyHFkJJkv4EPMIVcai7RXaGIpUp+0tOpt5vhcLkA=";
```
Comment (AndersonTorres) [inline pkgs/tools/networking/pgrok/default.nix:27]: Move this after `src`.

File: pkgs/tools/networking/pgrok/default.nix [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+  passthru.updateScript = nix-update-script { };
+
+  meta = with lib; {
```
Comment (AndersonTorres) [inline pkgs/tools/networking/pgrok/default.nix:31]:
```suggestion
  meta = {
```

File: pkgs/tools/networking/pgrok/default.nix [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    description = "Selfhosted TCP/HTTP tunnel, ngrok alternative, written in Go";
+    homepage = "https://github.com/pgrok/pgrok";
+    license = licenses.mit;
```
Comment (AndersonTorres) [inline pkgs/tools/networking/pgrok/default.nix:34]:
```suggestion
    license = lib.licenses.mit;
```

File: pkgs/tools/networking/pgrok/default.nix [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    maintainers = with maintainers; [ marie ];
```
Comment (AndersonTorres) [inline pkgs/tools/networking/pgrok/default.nix:35]:
```suggestion
    maintainers = with lib.maintainers; [ marie ];
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

**Comment by AndersonTorres** [comment]
@ofborg build pkgsLLVM.pgrok

**Review by AndersonTorres — APPROVED** [review]

**Review by drupol — APPROVED** [review]
Hi !

Thanks for your first contribution! Nothing to add in here, congrats :)

**Pull request merged**